CharlieBackend.Business/Services/LessonService.cs

```diff
@@ -145,10 +145,15 @@ public async Task<Result<Lesson>> AssignMentorToLessonAsync(AssignMentorToLesson
 
             if (mentorToAssign == null)
             {
-                throw new NullReferenceException();
+                return Result<Lesson>.GetError(ErrorCode.NotFound, $"Mentor with id {ids.MentorId} is not found");
             }
             var foundLesson = await _unitOfWork.LessonRepository.GetByIdAsync(ids.LessonId);
 
+            if (foundLesson == null)
+            {
+                return Result<Lesson>.GetError(ErrorCode.NotFound, $"Lesson with id {ids.LessonId} is not found");
```

Comment:
Please also update a corresponding controller method's summary to match this updated result.